webpack.common.config.js

```diff
@@ -81,9 +82,78 @@ module.exports = function(outputFilename) {
     // Webpack attempts to add a polyfill for process
     // and setImmediate, because q uses process to see
     // if it's in a Node.js environment
-    node: {
-      process: false,
-      setImmediate: false
-    }
+    node: false,
+
+    optimization: {
+      minimize: true,
+      minimizer: [
+        new TerserPlugin({
+          minify: (file, sourceMap) => {
+            let oktaLicenseBannersCount = 0;
+
+            // https://github.com/mishoo/UglifyJS2#minify-options
+            const uglifyJsOptions = {
+              compress: {
+                // Drop all console.* and Logger statements
+                drop_console: true,
+                drop_debugger: true,
+                pure_funcs: [
+                  'Logger.trace',
+                  'Logger.dir',
+                  'Logger.time',
+                  'Logger.timeEnd',
+                  'Logger.group',
+                  'Logger.groupEnd',
+                  'Logger.assert',
+                  'Logger.log',
+                  'Logger.info',
+                  'Logger.warn',
+                  'Logger.deprecate'
+                ],
+              },
+              sourceMap: true,
+              output: {
+                comments: (node, comment) => {
+                  // Remove other Okta copyrights
+                  const isLicense = /^!/.test(comment.value) ||
+                                  /.*(([Ll]icense)|([Cc]opyright)|(\([Cc]\))).*/.test(comment.value);
+                  const isOkta = /.*Okta.*/.test(comment.value);
+                  // Keep Okta license added by banner plugin - since it runs before optimization
+                  const isFirstOktaBanner = oktaLicenseBannersCount === 0;
+                  if (isOkta) {
+                    ++oktaLicenseBannersCount;
+                  }
+                  // Some licenses are in inline comments, rather than standard block comments.
+                  // UglifyJS2 treats consecutive inline comments as separate comments, so we
+                  // need exceptions to include all relevant licenses.
+                  const exceptions = [
+                    'Chosen, a Select Box Enhancer',
+                    'by Patrick Filler for Harvest',
+                    'Version 0.11.1',
+                    'Full source at https://github.com/harvesthq/chosen',
+          
+                    'Underscore.js 1.8.3'
+                  ];
+          
+                  const isException = exceptions.some(exception => {
+                    return comment.value.indexOf(exception) !== -1;
+                  });
+                  return (isLicense || isException) && (!isOkta || isFirstOktaBanner);
+                },
+              },
+              warnings: false
+            };
+  
+            if (sourceMap) {
+              uglifyJsOptions.sourceMap = {
+                content: sourceMap,
+              };
+            }
+  
+            return require('uglify-js').minify(file, uglifyJsOptions);
```

Comment:
Why are you using Uglify within Terser?

Reply:
there seem to be no clear way to specify license comments exceptions with Terser:
https://github.com/okta/okta-signin-widget/blob/8d9bbd5ff0e257e1391b3047750e02ee9b180c7c/scripts/buildtools/webpack/plugins.js#L64

Reply:
@sherizada-okta `uglify-js` removed